stylo: Bug 1353628 - Support shorthand properties for CSS Transition.

BorisChiou opened the pull request:
These patches add shorthand properties into TransitionProperty, so we can parse the shorthand properties properly and create transitions for them.

---
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix Bug 1353628
- [X] There are tests for these changes

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/16527)
<!-- Reviewable:end -->

BorisChiou committed:
Bug 1353628 - Part 1: Add shorthand properties into TransitionProperty.

In order to make TransitionProperty::parse() also work on shorthands, we
should add shorthands into TransitionProperty, and add the arms in other
functions which match TransitionProperty.

MozReview-Commit-ID: KFd26KcQf3N

BorisChiou committed:
Bug 1353628 - Part 2: Add unit tests of parsing transition properties.

MozReview-Commit-ID: K5thA20ORPk

BorisChiou committed:
Bug 1353628 - Part 3: Create PropertyAnimation for shorthands.

Therefore, we can start transitions on shorthand properties properly.

MozReview-Commit-ID: Ev2u2tEalnK

highfive commented:
Heads up! This PR modifies the following files:
 * @bholley: components/style/properties/properties.mako.rs, components/style/properties/helpers/animated_properties.mako.rs, components/style/gecko/wrapper.rs, components/style/animation.rs
 * @emilio: components/style/properties/properties.mako.rs, components/style/properties/helpers/animated_properties.mako.rs, components/style/gecko/wrapper.rs, components/style/animation.rs

BorisChiou commented:
@bors-servo r=emilio

bors-servo commented:
:pushpin: Commit 43f9f8c has been approved by `emilio`

<!-- @bors-servo r=emilio 43f9f8ca957a41f0e4dada0c33bf15d870be50a9 -->

bors-servo commented:
:hourglass: Testing commit 43f9f8ca957a41f0e4dada0c33bf15d870be50a9 with merge 3954ccdd0f0ec0978501c5498c8c1815abb7f29d...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/3504)

BorisChiou committed:
Bug 1353628 - Part 4: Update metadata for css-transition-1_dev.

1. All tests in events-001.htm are passed.
2. Parsing 'inherit' in transition-property-002.htm is passed.

BorisChiou commented:
@bors-servo r=emilio

Update metadata because some transition tests should be passed.

bors-servo commented:
:pushpin: Commit 8e90a2a has been approved by `emilio`

<!-- @bors-servo r=emilio 8e90a2a287844b510a39f3d5c745c79cab795ed6 -->

bors-servo commented:
:hourglass: Testing commit 8e90a2a287844b510a39f3d5c745c79cab795ed6 with merge 355d5f89da8fef0e8a2aaa94c45c2ff209e5ddae...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/6418), [arm32](http://build.servo.org/builders/arm32/builds/6633), [arm64](http://build.servo.org/builders/arm64/builds/6669), [linux-dev](http://build.servo.org/builders/linux-dev/builds/6768), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/3506), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/3531), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/6772), [mac-rel-css](http://build.servo.org/builders/mac-rel-css/builds/6811), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/3459), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/3533), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/1954)
Approved by: emilio
Pushing 355d5f89da8fef0e8a2aaa94c45c2ff209e5ddae to master...
